### PR TITLE
chore: fix some clippy warnings

### PR DESCRIPTION
**Pull request by Rhelvetican**
just some small fixes which hopefully shouldn't affect performance too much

### CODE REVIEW

File: src/utils/config.rs [hunk 1]
```diff
@@ -96,7 +98,7 @@ impl Config {
         DirBuilder::new()
             .recursive(true)
             .create(Path::new(&path))
-            .unwrap();
+            .expect("error creating directory.");
```
Comment (decipher3114) [inline src/utils/config.rs:101]: Change to Sentence Case

File: src/entities/app.rs [hunk 2]
```diff
@@ -10,11 +10,26 @@ use crate::entities::{
...
 
 pub struct App {
-    pub _tray_icon: TrayIcon,
+    #[expect(dead_code)]
+    tray_icon: TrayIcon,
     pub config: Config,
     pub windows: BTreeMap<Id, WindowType>,
 }
 
+impl App {
```
[the remark this answers is not in the document]
Comment (decipher3114) [reply src/entities/app.rs:19]: It is still not removed.

**Review by decipher3114 — CHANGES_REQUESTED** [review]
Make sure not replace unwraps for no reason. They'll soon be replaced with proper logic to handle exceptions.

File: src/utils/config.rs (hunk 1 above)
Comment (Rhelvetican) [reply src/utils/config.rs:101]: plan is to make a gracious error window instead but i am still researching iced

File: src/entities/app.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rhelvetican) [reply src/entities/app.rs:19]: the tray_icon field is never actually used, so I am thinking of removing it

[the remark this answers is not in the document]
Comment (Rhelvetican) [reply src/entities/app.rs:16]: Reconstructed.

[the remark this answers is not in the document]
Comment (decipher3114) [reply src/entities/app.rs:16]: Well, this field just prevents the icon from being dropped. It must live throughout the app execution.

**Review by decipher3114 — APPROVED** [review]

**Pull request merged**